Modules/Quotas/src/Quotas.Application/Infrastructure/Persistence/Repository/IIdentitiesRepository.cs

```diff
@@ -10,4 +11,6 @@ public interface IIdentitiesRepository
     Task Update(Identity identity, CancellationToken cancellationToken);
     Task<Identity> Find(string address, CancellationToken cancellationToken, bool track = false);
     Task<IEnumerable<Identity>> FindByAddresses(IReadOnlyCollection<string> identityAddresses, CancellationToken cancellationToken, bool track = false);
+    Task Delete(Identity identity, CancellationToken cancellationToken);
+    Task DeleteIdentities(Expression<Func<Identity, bool>> expression, CancellationToken cancellationToken);
```

Comment:
Isn't this just a `DeleteById`? If so, I don't think the use of delegates here serves any purpose other than make the code less readable.

Reply:
It was actually my suggestion. The thing is: if we use a `DeleteById`, we would need a `Find` method that gives us the ids. And there we will have the same problem like we do here.

We discussed a `DeleteByAddress` method or similar, but the decision on what to actually delete should be part of the Domain, and not of the Infrastructure.

Reply:
Imagine for example the Messages module. A Message should only be deleted if all participating Identities (sender and recipients) are deleted. If that's not the case, the Address of the deleted Identity should only be replaced by a "dummy".
This kind of logic definitely shouldn't be located in the Infrastructure